src/service.rs

```diff
@@ -23,13 +30,162 @@ impl Service for ApplicationService {
     type Parameters = ();
 
     async fn new(runtime: ServiceRuntime<Self>) -> Self {
-        let state = Application::load(runtime.root_view_storage_context())
-            .await
-            .expect("Failed to load state");
-        ApplicationService { state, runtime }
+        ApplicationService {
+            runtime: Arc::new(Mutex::new(runtime)),
+        }
+    }
+
+    async fn handle_query(&self, query: Self::Query) -> Self::QueryResponse {
+        Schema::build(
+            EmptyFields,
+            Mutation {
+                runtime: self.runtime.clone(),
+            },
+            EmptySubscription,
+        )
+        .finish()
+        .execute(query)
+        .await
+    }
+}
+
+/// Root type that defines all the GraphQL mutations available from the service.
+pub struct Mutation {
+    runtime: Arc<Mutex<ServiceRuntime<ApplicationService>>>,
+}
+
+#[async_graphql::Object]
+impl Mutation {
+    /// Executes a chat completion using the Atoma Network.
+    async fn chat(
+        &self,
+        api_token: String,
+        message: ChatMessage,
+    ) -> async_graphql::Result<Vec<u8>> {
+        let request = ChatCompletionRequest {
+            stream: false,
+            messages: &[&message],
+            model: "meta-llama/Llama-3.3-70B-Instruct".to_owned(),
+            max_tokens: 128,
+        };
+
+        let mut response = self.query_chat_completion(&api_token, &request)?;
+
+        let interaction = ChatInteractionResponse::parse_from_completion_response(response)?
+            .with_prompt(message.content);
+
+        Ok(
+            bcs::to_bytes(&Operation::LogChatInteraction { interaction })
+                .expect("`LogChatInteraction` should be serializable"),
+        )
+    }
+}
+
+/// A message to be sent to the AI chat.
+#[derive(Clone, Debug, Deserialize, Serialize, async_graphql::InputObject)]
+pub struct ChatMessage {
+    content: String,
+    role: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    author: Option<String>,
```

Comment:
I believe this field is usually referred to as `name`

Reply:
Fixed, thanks!